### DISCUSSION TITLE
tutor7pp which edep is edep

### DISCUSSION THREAD

**Discussion by Jaap1957** [nrc-cnrc/EGSnrc#825] [Questions] [answered]
Hello forum,

I try to use tutor7pp to find deposited energy. Here 3 maybe related questions about the exact meaning of edep.
In tutor7pp.cpp i meet Energy Deposited several times.

`line 456:        EGS_Float aux = the_epcont->edep*the_stack->wt[np];`
This looks very well as edep times weight. wt seems mostly 1.
question 1: why is wt needed here? 

`468:        aux = the_stack->wt[np]/the_stack->w[np];
469:        if (aux > 0) {` 

Before the binning, aux is redefined as weight / (z flight component)
question 2 : is this a bug? should not this be like:

`468:        float aux2 = the_stack->wt[np]/the_stack->w[np];
469:        if (aux2 > 0) {`

further on i see
`718:        EGS_Float edep = score->currentScore(ireg+1); `

Question 3: What is the relation of this 'score' edep to the  " the_epcont->edep " ?

Thanks for reading !

**Comment by rtownson** [comment]
1. The particle weight is always multiplied by deposited energy when energy deposition occurs. This scales a particles contribution to calculated quantities according to its weight, which is important for variance reduction techniques (e.g. particle splitting).
2. It's not a bug, maybe just unclear coding style. The change you suggest will not have any effect. The first value of `aux` is used at line 458. Then it is redefined at 468 before it is used with the new meaning.
3. `the_epcont->edep` is the energy deposited during a step in the simulation, e.g. one interaction event for one particle. `score->currentScore(ireg+1)` in that context is the total energy deposited to one particular region by an entire history (the primary and all its secondaries).

**Reply by Jaap1957** [reply]
Thanks  for anwering !

It still puzzles me a bit
The redefinition at 458 makes that in flu (eflu and gflu) the weight / w is scored instead of deposited energy.

the logging of eflu (lines 683 - 693)  is commented out, but i was interested in a radial energy fluence plot.

If I uncomment 683-693, I also expect I have to make sure edep * wt goes into eflu/gflu.

Or is scoring of wt / (cos(theta)) normally the way to get fluence distributions?

**Reply by rtownson** [reply]
The equation there currently calculates a special version of fluence (wt/w is essentially number of counts scaled by z-direction). If you would like to change it to energy fluence, multiply by the particle energy as well. Keep it mind that this would be calculating the deposited energy fluence, in a volume. It is an "estimated real fluence" because it is scaled by the direction of the particle - which is assuming that the beam is headed in the z-direction. This is not a conventional definition of fluence, which you normally might think of as the number of particles crossing an area, so make sure you understand the quantity that is actually being calculated and whether or not it is useful to you.

Note that you likely want to remove `ir = nreg+1;` at line 453. This was used for a special case of calculating transmitted/reflection fractions through a series of slabs.

**Reply by ftessier** [reply]
I have been revisiting the concept of fluence recently (again!), so I would like to comment in this. The 1/w factor (or 1/cosθ where θ is the angle with respect to the normal to the scoring plane) **_is_** the conventional definition of fluence, for a surface. It is there precisely to remove the dependence on angle; fluence is an isotropic quantity, i.e., it does not depend on the direction of the particle crossing the plane, as opposed to **_flux_**, which is the number of particles crossing the plane (per unit time).

A more fundamental definition of fluence is the average particle track length over the volume, for a given region. So one casual way to understand the cosθ factor is as follows. Consider then a very thin plate. When you tilt the plate, the track length increases ~1/cosθ. This factor is balanced by the number of tracks that hit the tilted plate, itself decreasing ~cosθ, so the fluence remains constant.

**Comment by Jaap1957** [comment, marked as answer]
Thank you Mr Townson !

I was aiming to get energy distribution as a function of R in a slab.
That, to find the point spread function of a direct electron camera.

Another distribution is the energy deposited per electorn in a slab.

The camera can view individual electrons hitting it.
I have put all energy deposited scores (so at the 'newshower' moment) in a file.

The distribution of grey values on our camera looks like a Landau curve:
like this example from the wiki landau page.

[cid:06ce6f90-4cc8-43dc-af35-096fdab9c31d]

My finding on the pulse height does not seem to have such a form.
At the left it keeps increasing until the cutoff energy is reached.

[cid:50bd5341-8193-4729-bf6c-d3ffbc42b632]

I do not expect the steep edge at the left.
It seems the "landau" part is underneath this curve, starting from 7 keV.
As I see it, 5keV is still very much > 3.6eV, the energy needed to free a e/p pair.
But on our real camera i do not see this peak at the left.

What do i do wrong ???

Jaap


________________________________

**Reply by rtownson** [reply]
To get energy deposited as a function of radius, create a geometry with radial rings (of the same material). Then score the energy deposited in each ring volume using the standard `egs_dose_scoring`. 

To get an energy deposition spectrum for each ring, you'd want a separate `EGS_ScoringArray` for each ring. Maybe this is what you're doing in post processing after you produce the file. In the Landau curve you're thinking of, is energy on the x-axis or radius? Of course many more individual energy depositions occur at low energies if you're counting them by number, as electrons slow down - is this what you're looking at? Your detector would not resolve those as individual counts. Instead, you want to look at the energy deposited by an entire history (per region). This would be done in `startNewShower` where you see `pheight[j]->score`.

**Reply by Jaap1957** [reply]
Thank you again !

The array per region is already there, I only have to check the "pulse height" part of tutor7pp.
Only, I do not understand the pulse height outcome.
line 723:   `pheight[j]->score(ibin,1);` : So, for each ibin found a "1" is scored.
line 677:    `f/ph_de[j],df/ph_de[j]);` 
prints #/particles per incoming particle in a certain bin, divided by bin width. so # of particles per MeV per incoming paricle.
Is that a useful pulse height parameter?

Anyway, to get the deposited energy in a certain bin, i changed
line 723:   `pheight[j]->score(ibin,edep);` 
line 677:    `f, f/dr);`    (relative error as last column)

Now f sums to the total regional dose in MeV per history.

**Reply by rtownson** [reply]
Make sure to watch out for `EGS_Float edep = score->currentScore(ireg+1);` depending on whether you are still using the special tutor7pp offset regions or not. Otherwise this seems ok.

Yes, the counts per MeV make sense. If you have a wider bin you will have more counts, and so if the bin widths are varying then it can be useful to look at the counts per unit bin width. This is a standard way of showing histograms.
